src/ndarray/ndarray_function.cc

```diff
@@ -26,35 +26,24 @@
 #include "./ndarray_function.h"
 #include "./ndarray_function-inl.h"
 #include "../common/utils.h"
-#include "../operator/mxnet_op.h"
 
 namespace mxnet {
 namespace ndarray {
 template<>
 void Copy<cpu, cpu>(const TBlob &from, TBlob *to,
                     Context from_ctx, Context to_ctx,
                     RunContext ctx) {
-  using namespace mxnet::op;
   MSHADOW_TYPE_SWITCH(to->type_flag_, DType, {
     if (to->type_flag_ == from.type_flag_) {
-      TBlob dest = to->FlatTo1D<cpu, DType>();
-      TBlob src = from.FlatTo1D<cpu, DType>();
-      const size_t size = src.Size();
-      if (dest.CheckContiguous() && src.CheckContiguous() && size >= 20000 /* non-trivial size */) {
-        CHECK_EQ(dest.shape_, src.shape_)
-          << "Copy:shape mismatch:" << dest.shape_ << " vs " << src.shape_;
-          mxnet_op::Kernel<mxnet_op::op_with_req<mshadow_op::identity, kWriteTo>, cpu>::Launch(
-            ctx.get_stream<cpu>(), src.Size(), dest.dptr<DType>(), src.dptr<DType>());
```

Comment:
So we cannot use kernel::launch together with shared memory?

Reply:
I don't see why not, but I am just going to revert now because the change wasn't worth the trouble it caused.

Reply:
It does get stuck in the kernel launch code for whatever reason.

Reply:
Yes, I think we can revert first and find the cause later. What do you think @marcoabreu . This error is really critical as the DataLoader is really common among users.

Reply:
I think the fork doesn't play nicely with libgomp in this codepath because it works fine when linked against mkl, which uses libiomp.  Or if built with cmake, which builds and links against Intel omp (in 3rdparty), it works fine.

Reply:
Using the keyword I find this link: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=58378

Reply:
Ha! Nice find! That's exactly it.

Reply:
It suggests start the multiprocessing using `forkserver` (https://docs.python.org/3/library/multiprocessing.html#multiprocessing.set_start_method). I'm trying this keyword.

Reply:
Intel OMP at fork, sets all its init vars to uninitialized, so that I suppose it recreates its state in the new fork:

https://github.com/pmodels/bolt/blob/master/runtime/src/z_Linux_util.c#L1536

Reply:
Looks like someone tried to fix it once: https://patchwork.ozlabs.org/patch/319827/

Reply:
Failed to use `forkserver`. I'll continue working on something else.

Reply:
@sxjscience sure, go ahead and merge. This seems pretty critical